lib/scarpe/promises.rb

```diff
@@ -283,9 +285,8 @@ def call_handlers_for(state)
           begin
             @scheduler.call(*@parents.map(&:returned_value))
           rescue => e
-            if Promise.debug
-              puts "Error while running scheduler! #{e.full_message}"
-            end
+            caller_class = caller_locations(1, 1)[0].label
```

Comment:
is this is what you meant? to use caller... instead of promise.debug i am not sure about this so yeah